# Comparing `tmp/pciSeq-0.0.52rc0.dev0-py3-none-any.whl.zip` & `tmp/pciSeq-0.0.52rc1.dev0-py3-none-any.whl.zip`

## zipinfo {}

```diff
@@ -1,14 +1,14 @@
-Zip file size: 364260 bytes, number of entries: 73
+Zip file size: 364321 bytes, number of entries: 73
 -rw-rw-rw-  2.0 fat     2135 b- defN 23-Jun-25 20:54 pciSeq/__init__.py
 -rw-rw-rw-  2.0 fat    10910 b- defN 23-Jun-29 10:21 pciSeq/app.py
 -rw-rw-rw-  2.0 fat     3367 b- defN 23-Jun-29 10:21 pciSeq/config.py
 -rw-rw-rw-  2.0 fat     3678 b- defN 22-Apr-07 23:15 pciSeq/make_plot.py
 -rw-rw-rw-  2.0 fat        0 b- defN 22-Jan-19 17:53 pciSeq/src/__init__.py
--rw-rw-rw-  2.0 fat       29 b- defN 23-Jul-02 21:05 pciSeq/src/_version.py
+-rw-rw-rw-  2.0 fat       30 b- defN 23-Jul-02 22:45 pciSeq/src/_version.py
 -rw-rw-rw-  2.0 fat        0 b- defN 22-Jan-19 17:53 pciSeq/src/cell_call/__init__.py
 -rw-rw-rw-  2.0 fat    20864 b- defN 23-Jun-29 10:21 pciSeq/src/cell_call/datatypes.py
 -rw-rw-rw-  2.0 fat     2995 b- defN 23-Apr-26 22:13 pciSeq/src/cell_call/log_config.py
 -rw-rw-rw-  2.0 fat    18989 b- defN 23-Jul-02 11:57 pciSeq/src/cell_call/main.py
 -rw-rw-rw-  2.0 fat     3104 b- defN 23-Jun-25 20:54 pciSeq/src/cell_call/summary.py
 -rw-rw-rw-  2.0 fat     9194 b- defN 23-Jun-25 20:54 pciSeq/src/cell_call/utils.py
 -rw-rw-rw-  2.0 fat        0 b- defN 23-Jun-11 12:10 pciSeq/src/diagnostics/__init__.py
@@ -62,14 +62,14 @@
 -rw-rw-rw-  2.0 fat    15374 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/lib/js/leaflet-pip.js
 -rw-rw-rw-  2.0 fat     5927 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/lib/js/leaflet.textpath.js
 -rw-rw-rw-  2.0 fat      265 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/lib/js/preloader.js
 -rw-rw-rw-  2.0 fat     9097 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/lib/js/pixiOverlay/MarkerContainer.js
 -rw-rw-rw-  2.0 fat     4434 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/lib/js/pixiOverlay/bezier-easing.js
 -rw-rw-rw-  2.0 fat   613808 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/lib/js/pixiOverlay/example.min.js
 -rw-rw-rw-  2.0 fat    91874 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/lib/js/pixiOverlay/tools.min.js
--rw-rw-rw-  2.0 fat     1091 b- defN 23-Jul-02 21:05 pciSeq-0.0.52rc0.dev0.dist-info/LICENCE
--rw-rw-rw-  2.0 fat     4119 b- defN 23-Jul-02 21:05 pciSeq-0.0.52rc0.dev0.dist-info/METADATA
--rw-rw-rw-  2.0 fat       92 b- defN 23-Jul-02 21:05 pciSeq-0.0.52rc0.dev0.dist-info/WHEEL
--rw-rw-rw-  2.0 fat       48 b- defN 23-Jul-02 21:05 pciSeq-0.0.52rc0.dev0.dist-info/entry_points.txt
--rw-rw-rw-  2.0 fat        7 b- defN 23-Jul-02 21:05 pciSeq-0.0.52rc0.dev0.dist-info/top_level.txt
-?rw-rw-r--  2.0 fat     6983 b- defN 23-Jul-02 21:05 pciSeq-0.0.52rc0.dev0.dist-info/RECORD
-73 files, 1325027 bytes uncompressed, 352914 bytes compressed:  73.4%
+-rw-rw-rw-  2.0 fat     1091 b- defN 23-Jul-02 22:47 pciSeq-0.0.52rc1.dev0.dist-info/LICENCE
+-rw-rw-rw-  2.0 fat     4287 b- defN 23-Jul-02 22:47 pciSeq-0.0.52rc1.dev0.dist-info/METADATA
+-rw-rw-rw-  2.0 fat       92 b- defN 23-Jul-02 22:47 pciSeq-0.0.52rc1.dev0.dist-info/WHEEL
+-rw-rw-rw-  2.0 fat       48 b- defN 23-Jul-02 22:47 pciSeq-0.0.52rc1.dev0.dist-info/entry_points.txt
+-rw-rw-rw-  2.0 fat        7 b- defN 23-Jul-02 22:47 pciSeq-0.0.52rc1.dev0.dist-info/top_level.txt
+-rw-rw-r--  2.0 fat     6983 b- defN 23-Jul-02 22:47 pciSeq-0.0.52rc1.dev0.dist-info/RECORD
+73 files, 1325196 bytes uncompressed, 352975 bytes compressed:  73.4%
```

## zipnote {}

```diff
@@ -195,26 +195,26 @@
 
 Filename: pciSeq/static/2D/viewer/js/lib/js/pixiOverlay/example.min.js
 Comment: 
 
 Filename: pciSeq/static/2D/viewer/js/lib/js/pixiOverlay/tools.min.js
 Comment: 
 
-Filename: pciSeq-0.0.52rc0.dev0.dist-info/LICENCE
+Filename: pciSeq-0.0.52rc1.dev0.dist-info/LICENCE
 Comment: 
 
-Filename: pciSeq-0.0.52rc0.dev0.dist-info/METADATA
+Filename: pciSeq-0.0.52rc1.dev0.dist-info/METADATA
 Comment: 
 
-Filename: pciSeq-0.0.52rc0.dev0.dist-info/WHEEL
+Filename: pciSeq-0.0.52rc1.dev0.dist-info/WHEEL
 Comment: 
 
-Filename: pciSeq-0.0.52rc0.dev0.dist-info/entry_points.txt
+Filename: pciSeq-0.0.52rc1.dev0.dist-info/entry_points.txt
 Comment: 
 
-Filename: pciSeq-0.0.52rc0.dev0.dist-info/top_level.txt
+Filename: pciSeq-0.0.52rc1.dev0.dist-info/top_level.txt
 Comment: 
 
-Filename: pciSeq-0.0.52rc0.dev0.dist-info/RECORD
+Filename: pciSeq-0.0.52rc1.dev0.dist-info/RECORD
 Comment: 
 
 Zip file comment:
```

## pciSeq/src/_version.py

```diff
@@ -1 +1 @@
-__version__ = '0.0.52.rc.dev'
+__version__ = '0.0.52.rc1.dev'
```

## Comparing `pciSeq-0.0.52rc0.dev0.dist-info/LICENCE` & `pciSeq-0.0.52rc1.dev0.dist-info/LICENCE`

 * *Files identical despite different names*

## Comparing `pciSeq-0.0.52rc0.dev0.dist-info/METADATA` & `pciSeq-0.0.52rc1.dev0.dist-info/METADATA`

 * *Files 13% similar despite different names*

```diff
@@ -1,93 +1,96 @@
-Metadata-Version: 2.1
-Name: pciSeq
-Version: 0.0.52rc0.dev0
-Summary: Probabilistic cell typing for spatial transcriptomics
-Home-page: https://github.com/acycliq/pciSeq
-Author: Dimitris Nicoloutsopoulos
-Author-email: dimitris.nicoloutsopoulos@gmail.com
-License: BSD
-Classifier: Programming Language :: Python :: 3
-Classifier: License :: OSI Approved :: BSD License
-Classifier: Operating System :: OS Independent
-Description-Content-Type: text/markdown
-License-File: LICENCE
-Requires-Dist: numpy-groupies
-Requires-Dist: pandas
-Requires-Dist: scipy
-Requires-Dist: scikit-image
-Requires-Dist: scikit-learn
-Requires-Dist: tqdm
-Requires-Dist: flask
-Requires-Dist: numexpr
-Requires-Dist: diplib
-Requires-Dist: pyvips
-Requires-Dist: natsort
-Requires-Dist: redis
-Provides-Extra: interactive
-Requires-Dist: matplotlib (>=2.2.0) ; extra == 'interactive'
-Requires-Dist: jupyter ; extra == 'interactive'
-
-# pciSeq: Probabilistic Cell typing by In situ Sequencing
-A Python package that implements the cell calling algorithm as described in [Qian, X., et al. Nature Methods (2020)](https://www.nature.com/articles/s41592-019-0631-4)
-<p align="center">
-    <img src="https://github.com/acycliq/pciSeq/blob/master/assets/screencast_resized.gif?raw=true" alt="screenshot"/>
-</p>
-
-## Installation
-```
-python -m pip install pciSeq
-```
-Requirement: Python >= 3.8
-
-If you want to work with the source code you can download the repo and then replicate the python environment by
-```
-conda env create -n pciSeq -f /path/to/environment.yml
-```
-
-That will create a conda environment with the name `pciSeq` containing all the necessary packages to run the algorithm. To activate it run 
-```
-conda activate pciSeq
-```
-or, if you open the project in your IDE, then in your project settings, switch your interpreter to the interpreter of the `pciSeq` env. 
-## Usage
-You need to create two `pandas dataframes` for the spots and the single cell data and a `coo_matrix` for the label image (which in 
-most cases will be the output of some image segmentation application). Then you pass them into the `pciSeq.fit()` method as follows: 
-```
-import pciSeq
-
-res = pciSeq.fit(spots=spots_df, coo=label_image, scRNAseq=scRNA_df)
-```
-See the demo below for a more detailed explanation about the arguments of  `pciSeq.fit()` and its return values.
-
-There is also a fourth argument (optional) to override the default hyperparameter values which are initialised 
-by the [config.py](https://github.com/acycliq/pciSeq/blob/master/pciSeq/config.py) module. To pass user-defined hyperparameter values, create a `dictionary` with `keys` the
-hyperparameter names and `values` their new values. For example, to exclude all Npy and Vip spots you can do:
-
-```
-import pciSeq
-
-opts = { 'exclude_genes': ['Npy', 'Vip'] }
-res = pciSeq.fit(spots=spots_df, coo=label_image, scRNAseq=scRNA_df, opts=opts)
-```
-
-## Demo
-You can run a pciSeq demo in google colab: [![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/acycliq/pciSeq/blob/master/notebooks/1_pciSeq.ipynb)
-
-## Viewer
-An interactive viewer to explore the data runs on this [url](https://acycliq.github.io/visage/). Instructions about 
-building this viewer with your own data are [here](https://github.com/acycliq/visage). \
-If you have `v 0.0.49` or greater you can also launch the viewer automatically by 
-setting `opts = {'launch_viewer': True}` and passing it to `pciSeq.fit()`, see [![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/acycliq/pciSeq/blob/master/notebooks/2_viewer.ipynb)
-
-## Change Log
-### [0.0.50] - 2023-05-27
- - Single cell data are optional, more info can be found here [![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/acycliq/pciSeq/blob/master/notebooks/3_pciSeq_without_singleCell_data.ipynb)
-
- - `pciSeq.fit()` takes keyword arguments
-
-
-## References 
-Qian, X., et al. (2020). Probabilistic cell typing enables fine mapping of closely related cell types in situ. Nat
-Methods 17, 101 - 106.
-
-
+Metadata-Version: 2.1
+Name: pciSeq
+Version: 0.0.52rc1.dev0
+Summary: Probabilistic cell typing for spatial transcriptomics
+Home-page: https://github.com/acycliq/pciSeq
+Author: Dimitris Nicoloutsopoulos
+Author-email: dimitris.nicoloutsopoulos@gmail.com
+License: BSD
+Classifier: Programming Language :: Python :: 3
+Classifier: License :: OSI Approved :: BSD License
+Classifier: Operating System :: OS Independent
+Description-Content-Type: text/markdown
+License-File: LICENCE
+Requires-Dist: numpy-groupies
+Requires-Dist: pandas
+Requires-Dist: scipy
+Requires-Dist: streamlit
+Requires-Dist: altair
+Requires-Dist: scikit-image
+Requires-Dist: scikit-learn
+Requires-Dist: tqdm
+Requires-Dist: flask
+Requires-Dist: numexpr
+Requires-Dist: diplib
+Requires-Dist: pyvips
+Requires-Dist: natsort
+Requires-Dist: redis
+Provides-Extra: interactive
+Requires-Dist: matplotlib (>=2.2.0) ; extra == 'interactive'
+Requires-Dist: jupyter ; extra == 'interactive'
+
+# pciSeq: Probabilistic Cell typing by In situ Sequencing
+A Python package that implements the cell calling algorithm as described in [Qian, X., et al. Nature Methods (2020)](https://www.nature.com/articles/s41592-019-0631-4)
+<p align="center">
+    <img src="https://github.com/acycliq/pciSeq/blob/master/assets/screencast_resized.gif?raw=true" alt="screenshot"/>
+</p>
+
+## Installation
+```
+python -m pip install pciSeq
+```
+Requirement: Python >= 3.8
+
+If you want to work with the source code you can download the repo and then replicate the python environment by
+```
+conda env create -n pciSeq -f /path/to/environment.yml
+```
+
+That will create a conda environment with the name `pciSeq` containing all the necessary packages to run the algorithm. To activate it run 
+```
+conda activate pciSeq
+```
+or, if you open the project in your IDE, then in your project settings, switch your interpreter to the interpreter of the `pciSeq` env. 
+## Usage
+You need to create two `pandas dataframes` for the spots and the single cell data and a `coo_matrix` for the label image (which in 
+most cases will be the output of some image segmentation application). Then you pass them into the `pciSeq.fit()` method as follows: 
+```
+import pciSeq
+
+res = pciSeq.fit(spots=spots_df, coo=label_image, scRNAseq=scRNA_df)
+```
+See the demo below for a more detailed explanation about the arguments of  `pciSeq.fit()` and its return values.
+
+There is also a fourth argument (optional) to override the default hyperparameter values which are initialised 
+by the [config.py](https://github.com/acycliq/pciSeq/blob/master/pciSeq/config.py) module. To pass user-defined hyperparameter values, create a `dictionary` with `keys` the
+hyperparameter names and `values` their new values. For example, to exclude all Npy and Vip spots you can do:
+
+```
+import pciSeq
+
+opts = { 'exclude_genes': ['Npy', 'Vip'] }
+res = pciSeq.fit(spots=spots_df, coo=label_image, scRNAseq=scRNA_df, opts=opts)
+```
+
+## Demo
+You can run a pciSeq demo in google colab: [![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/acycliq/pciSeq/blob/master/notebooks/1_pciSeq.ipynb)
+
+## Viewer
+An interactive viewer to explore the data runs on this [url](https://acycliq.github.io/visage/). Instructions about 
+building this viewer with your own data are [here](https://github.com/acycliq/visage). \
+If you have `v 0.0.49` or greater you can also launch the viewer automatically by 
+setting `opts = {'launch_viewer': True}` and passing it to `pciSeq.fit()`, see [![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/acycliq/pciSeq/blob/master/notebooks/2_viewer.ipynb)
+
+## Diagnostics dashboard
+## Change Log
+### [0.0.50] - 2023-05-27
+ - Single cell data are optional, more info can be found here [![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/acycliq/pciSeq/blob/master/notebooks/3_pciSeq_without_singleCell_data.ipynb)
+
+ - `pciSeq.fit()` takes keyword arguments
+
+
+## References 
+Qian, X., et al. (2020). Probabilistic cell typing enables fine mapping of closely related cell types in situ. Nat
+Methods 17, 101 - 106.
+
+
```

## Comparing `pciSeq-0.0.52rc0.dev0.dist-info/RECORD` & `pciSeq-0.0.52rc1.dev0.dist-info/RECORD`

 * *Files 2% similar despite different names*

```diff
@@ -1,13 +1,13 @@
 pciSeq/__init__.py,sha256=mxqjE8smLqDQuwzheJ_iA3gfcyV7c4x8lHQPwSEQwJk,2135
 pciSeq/app.py,sha256=WTBF8N7hbVz6DE3yroUgXL23ophYgyQno_s5V-A0MF4,10910
 pciSeq/config.py,sha256=DFgt5LpDUBlLNQNecM-TfoTry3388LbdRU4jAxhxDHA,3367
 pciSeq/make_plot.py,sha256=2AkncrVVvytr2keK505oMC0LV1Txli9WpsSGpjmht_4,3678
 pciSeq/src/__init__.py,sha256=47DEQpj8HBSa-_TImW-5JCeuQeRkm5NMpJWZG3hSuFU,0
-pciSeq/src/_version.py,sha256=sbWsm0Hlx0ncNOGSZ5Po_YseZhVPUVhv0UwnE8epnec,29
+pciSeq/src/_version.py,sha256=HO0UUAdY7MxW1laIZs0AzIDVXE_O48c2DGtQC2BT-mg,30
 pciSeq/src/cell_call/__init__.py,sha256=47DEQpj8HBSa-_TImW-5JCeuQeRkm5NMpJWZG3hSuFU,0
 pciSeq/src/cell_call/datatypes.py,sha256=om8kMxOUxKcAoQHSTgXEuX4UNMGtwTjFkOCGmeublqs,20864
 pciSeq/src/cell_call/log_config.py,sha256=vnm9TxlMEVG_GLPfl21GDEWTafEk3B-drTLsVQMfFEE,2995
 pciSeq/src/cell_call/main.py,sha256=Vi4PvkIrljvyLpgK6AagXBdcreApX8uFkAqqbbVDQ7c,18989
 pciSeq/src/cell_call/summary.py,sha256=f2snLqJIeywtYuHrJLM-hRvifgqRU2u2aCKE_JrZJPs,3104
 pciSeq/src/cell_call/utils.py,sha256=M4KJrS82pIHuCoxg2ej6NE9fJAVCvJw90T7YmWI2T7o,9194
 pciSeq/src/diagnostics/__init__.py,sha256=47DEQpj8HBSa-_TImW-5JCeuQeRkm5NMpJWZG3hSuFU,0
@@ -61,13 +61,13 @@
 pciSeq/static/2D/viewer/js/lib/js/leaflet-pip.js,sha256=QqsilU0SHTaKA_qNSv_jEDH53cWO6T_RdforfybnbKM,15374
 pciSeq/static/2D/viewer/js/lib/js/leaflet.textpath.js,sha256=i5cmMPqpqXShZz0TvXU3ZQUi2Y07SIbIpMYbhPE8gWE,5927
 pciSeq/static/2D/viewer/js/lib/js/preloader.js,sha256=eaFWRf9yT-Td4C2t01P8fDEZYszYTaGDgQ1vjevNIDI,265
 pciSeq/static/2D/viewer/js/lib/js/pixiOverlay/MarkerContainer.js,sha256=Z8TADCqYEQr25OEF7wljsOVAOpx87UrjFPI2nvIty70,9097
 pciSeq/static/2D/viewer/js/lib/js/pixiOverlay/bezier-easing.js,sha256=yMoRDqJiglg0M0eWM25WGA5zzLWuJpOyC3a8aGFx0jw,4434
 pciSeq/static/2D/viewer/js/lib/js/pixiOverlay/example.min.js,sha256=bZEnK8XSnShrpLYtOS7Xq7wnp-nW_nMZh4aaDc1Gt6U,613808
 pciSeq/static/2D/viewer/js/lib/js/pixiOverlay/tools.min.js,sha256=A027Ub-oWXvKbz2B3pgMFS5xApmnEHXz9GQh79YZZnU,91874
-pciSeq-0.0.52rc0.dev0.dist-info/LICENCE,sha256=6kbiFSfobTZ7beWiKnHpN902HgBx-Jzgcme0SvKqhKY,1091
-pciSeq-0.0.52rc0.dev0.dist-info/METADATA,sha256=EBiFhaJj2ful1qOoOvZoC5Ynnzt19YAh42TtTR8hEFc,4119
-pciSeq-0.0.52rc0.dev0.dist-info/WHEEL,sha256=G16H4A3IeoQmnOrYV4ueZGKSjhipXx8zc8nu9FGlvMA,92
-pciSeq-0.0.52rc0.dev0.dist-info/entry_points.txt,sha256=RGOkAd0S7rrTuEInj-jKu5B7630i4R3mK9v9vx6nqc8,48
-pciSeq-0.0.52rc0.dev0.dist-info/top_level.txt,sha256=mSonMN0SaB51m5yLGm5hFE1Fthf-2U4sRKxSezY-JMk,7
-pciSeq-0.0.52rc0.dev0.dist-info/RECORD,,
+pciSeq-0.0.52rc1.dev0.dist-info/LICENCE,sha256=6kbiFSfobTZ7beWiKnHpN902HgBx-Jzgcme0SvKqhKY,1091
+pciSeq-0.0.52rc1.dev0.dist-info/METADATA,sha256=PDQIMYxMAcM2H7T0Hm6Ge1Pm95JnboLby2xVgPA0Scg,4287
+pciSeq-0.0.52rc1.dev0.dist-info/WHEEL,sha256=pkctZYzUS4AYVn6dJ-7367OJZivF2e8RA9b_ZBjif18,92
+pciSeq-0.0.52rc1.dev0.dist-info/entry_points.txt,sha256=RGOkAd0S7rrTuEInj-jKu5B7630i4R3mK9v9vx6nqc8,48
+pciSeq-0.0.52rc1.dev0.dist-info/top_level.txt,sha256=mSonMN0SaB51m5yLGm5hFE1Fthf-2U4sRKxSezY-JMk,7
+pciSeq-0.0.52rc1.dev0.dist-info/RECORD,,
```

